src/query/service/src/sessions/query_ctx.rs

```diff
@@ -498,7 +499,7 @@ impl TableContext for QueryContext {
         catalog_name: &str,
         database_name: &str,
         table_name: &str,
-        files: Vec<StageFileInfo>,
+        mut files: Vec<StageFileInfo>,
```

Comment:
We can make a copy, but please don't make the argument to `mut`. This is bit tricky.